activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -323,6 +322,8 @@ def initialize(spec)
 
         @spec = spec
 
+        @handler = ActiveRecord::Base.connection_handler
```

Comment:
Can we assume that all models share the same connection handler than `ActiveRecord::Base`?